dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -2093,47 +2098,52 @@ public void deleteWorkflow(@ApiParam(hidden = true) @Parameter(hidden = true, na
+        @Parameter(name = "X-GitHub-Delivery", in = ParameterIn.HEADER, description = "A GUID to identify the GitHub webhook delivery", required = true) @HeaderParam(value = "X-GitHub-Delivery")  String deliveryId,
+        @RequestBody(description = "GitHub push event payload", required = true) String payload) {
+        final GHEventPayload.Push pushPayload = parseEventPayload(payload, GHEventPayload.Push.class);
+        final long installationId = getGitHubInstallationId(pushPayload);
+        final String username = getGitHubUsername(pushPayload);
+        final String repository = pushPayload.getRepository().getFullName();
+        final String gitReference = pushPayload.getRef();
+
         if (LOG.isInfoEnabled()) {
             LOG.info(String.format("Branch/tag %s pushed to %s(%s)", Utilities.cleanForLogging(gitReference), Utilities.cleanForLogging(repository), Utilities.cleanForLogging(username)));
         }
-        final String deliveryId = LambdaEvent.createDeliveryId();
-        githubWebhookRelease(repository, username, gitReference, Long.parseLong(installationId), deliveryId, true);
+
+        githubWebhookRelease(repository, username, gitReference, installationId, deliveryId, true);
     }
 
     @POST
     @Path("/github/install")
     @Timed
-    @Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+    @Consumes(MediaType.APPLICATION_JSON)
     @UnitOfWork
     @RolesAllowed({"curator", "admin"})
     @Operation(description = "Handle the installation of our GitHub app onto a repository or organization.", security = @SecurityRequirement(name = JWT_SECURITY_DEFINITION_NAME), responses = @ApiResponse(responseCode = "418", description = "This code tells AWS Lambda not to retry."))
     @ApiOperation(value = "Handle the installation of our GitHub app onto a repository or organization.", authorizations = {
         @Authorization(value = JWT_SECURITY_DEFINITION_NAME)}, response = Workflow.class, responseContainer = "List")
     public Response handleGitHubInstallation(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user") @Auth User user,
-        @Parameter(name = "repositories", description = "Comma-separated repository paths (ex. dockstore/dockstore-ui2) for all repositories installed", required = true) @FormParam("repositories") String repositories,
-        @Parameter(name = "username", description = "Username of user on GitHub who triggered action", required = true) @FormParam("username") String username,
-        @Parameter(name = "installationId", description = "GitHub installation ID", required = true) @FormParam("installationId") String installationId) {
+            @Parameter(name = "X-GitHub-Delivery", in = ParameterIn.HEADER, description = "A GUID to identify the GitHub webhook delivery", required = true) @HeaderParam(value = "X-GitHub-Delivery")  String deliveryId,
```

Comment:
I commented in your lambda PR about this, but I think there's a tiny window after deployment where in theory deliveryId could be undefined. That means some events could be lost.

You could just generate one in the lambda if it's not present and leave this code as is.